OD_data_cleaning/OD_data_cleaning.py

```diff
@@ -0,0 +1,123 @@
+### Libraries ###
+import pandas as pd
+from datetime import datetime
+import numpy as np
+import croissance
+import re
+
+# ------ DATA LOADING AND VARIABLE SELECTION ------
+
+#Open the file
+df_raw = pd.read_excel("results5.xlsx")
+
+#We select the valuable columns
+cols = ["Sample ID", "Measurement", "Measurement type", "Sampling time","Sampling date"] #Include dilution rate ?
+df_raw = df_raw[cols]
+df_raw.columns = df_raw.columns.str.replace(' ', '_')			#replace white spaces on headers by underscores
+
+''' This is a piece of code that generates the data
+#We create a dataframe with the purpose of every sample (this is just for tests, known input)
+df_raw["calc_volumeloss"] = df_raw.Sample_ID.str.contains(r"\S{3}_[A-H]12")
+df_raw["calc_gr"] = ~df_raw.Sample_ID.str.contains(r"\S{3}_[A-H]12")    
+df_test = pd.DataFrame()
+
+df_test = df_raw[["Sample_ID", "calc_gr","calc_volumeloss"]]
+
+df_test = df_test.drop_duplicates() 
+df_test.to_csv('calc.tsv', sep = '\t')
```

Comment:
Are these Disk IO operations needed?  As far as I can see, you can just pass the data frame to the next processes without writing to Disk.  Unfortunately, Disk IO is quite slow and should be minimized if possible.